onadata/apps/api/viewsets/connect_viewset.py

```diff
@@ -114,6 +122,80 @@ def regenerate_auth_token(self, request, *args, **kwargs):
 
         return Response(data=new_token.key, status=status.HTTP_201_CREATED)
 
+    @action(methods=['GET', 'POST', 'PATCH'], detail=False)
+    def odk_token(self, request, *args, **kwargs):
+        user = request.user
+        acceptable_statuses = [ODKToken.ACTIVE, ODKToken.INACTIVE]
+
+        if request.method == 'GET':
+            try:
+                token = ODKToken.objects.get(user=user)
+                expiry_date = token.created + timedelta(
+                    days=ODK_TOKEN_LIFETIME)
+                return Response(data={
+                    'enc_odk_token': token.key,
+                    'active_till': expiry_date,
+                    'status': token.status,
+                    'status_display': token.get_status_display()
```

Comment:
Stick to `'status': 'ACTIVE | INACTIVE'`. Or simply do not return it, there is no meaningful use of the status since we should only return an active token.

Reply:
Chose to simply not return the status information. The changes have been implemented in the latest commits.